### PR TITLE
Extract more data from FERC XBRLs and handle that new data in ETL

### PR DESCRIPTION
**Pull request by jdangerx**
Changes required to get the FERC 1 assets materializing properly, while pointed at the new XBRL extractor (for a more-complete xbrl2sqlite) and the new XBRL archives.

* change how we find the filings to run `xbrl2sqlite` on
* update how we call `ferc-xbrl-extractor` - to match what's on the `api_compat` branch
* change `num_transmission_circuits` from `int` to `number` - this value contains `"0.0"` which `int` doesn't like, but `float` can handle. Potentially we can do a pre-conversion from string to float, and then convert that to int when we're applying dtypes. This was breaking `transmission_statistics_ferc1`.
* drop end-of-previous-year values for the instant table that feeds `utility_plant_summary_ferc1`
* add `total` and `other` as utility type categories across the board. These were being turned into `NA`s because they were "unrecognizable" categories, which then led to spurious dupes down the line, and broke `electric_plant_depreciation_changes_ferc1`.

One big change we made in the extractor itself was to take multiple filings from one entity and merge them, treating later filings as updates to earlier filings.

TODO:
* [x] remove "grab last filing only" logic - this should be obviated by the "take multiple filings and merge them" logic mentioned above.
  * [ ] [we do still have to refactor to pass the report publish time into `ferc-xbrl-extractor` instead of relying on the `ReportDate` fact, but I think that can be a follow-up PR.](https://github.com/catalyst-cooperative/pudl/issues/2822)
* [x] fix unit tests that touch our `extract.xbrl` logic, since we changed the logic quite a lot
* [x] release `api-compat` as ferc-xbrl-extractor 1.0, and point PUDL at that instead of this git ref we have now
* [x] see if we can standardize on new-style or old-style archives - this code is currently written to handle "new-style" archives that have the taxonomies separately archived from the XBRL files, but all archives except for 10.5072/zenodo.1234455 are still old-style.

### CODE REVIEW

File: src/pudl/metadata/fields.py [hunk 1]
```diff
@@ -1552,7 +1552,7 @@
         "description": "For nuclear plants only, the unit number .One digit numeric. Nuclear plants are the only type of plants for which data are shown explicitly at the generating unit level.",
     },
     "num_transmission_circuits": {
-        "type": "integer",
+        "type": "number",
```
Comment (zaneselvans) [inline src/pudl/metadata/fields.py:1555]: Is it possible to have a non-integer number of transmission circuits?  If not, then I think we probably want to transform the input strings such that they're compatible with integer types. We've done this elsewhere by using something like:

```py
df["my_int_col"] = pd.to_numeric(df.floaty_int_strings).astype(int)
```

`pd.to_numeric()` will happily go from string to float, and assuming all the floats can be represented as ints, the `astype()` will be successful.

File: src/pudl/transform/ferc1.py [hunk 2]
```diff
@@ -4383,8 +4359,17 @@ def transform_main(self: Self, df: pd.DataFrame) -> pd.DataFrame:
...
+            (year, 211, utility_type, column_name)
+            for year in [2006] + list(range(2009, 2021))
+            for utility_type in ["electric", "total"]
+            for column_name in [
+                "accumulated_provision_for_depreciation_amortization_and_depletion_of_plant_utility",
+                "amortization_of_other_utility_plant_utility_plant_in_service",
+                "depreciation_amortization_and_depletion_utility_plant_in_service",
+                "depreciation_utility_plant_in_service",
+            ]
+        ]
```
Comment (zaneselvans) [inline src/pudl/transform/ferc1.py:4372]: Oh that's much nicer.

File: pyproject.toml [hunk 3]
```diff
@@ -14,7 +14,7 @@ dependencies = [
     "addfips>=0.4,<0.5",
     "alembic>=1.10.3,<1.13",
     "catalystcoop.dbfread>=3.0,<3.1",
-    "catalystcoop.ferc-xbrl-extractor==0.8.3",
+    "catalystcoop.ferc-xbrl-extractor>=1.0.0,<2",
```
Comment (zaneselvans) [inline pyproject.toml:17]: We probably want to require at least `v1.1.0` to avoid the memory issue don't we?

File: src/pudl/output/ferc1.py
Comment (jdangerx) [inline src/pudl/output/ferc1.py:1]: TODO: this shouldn't have to be 1.00 anymore.

File: src/pudl/output/ferc714.py
Comment (jdangerx) [inline src/pudl/output/ferc714.py:1]: Just some type stuff.

File: src/pudl/transform/classes.py [hunk 4]
```diff
@@ -817,7 +817,12 @@ def drop_invalid_rows(df: pd.DataFrame, params: InvalidRows) -> pd.DataFrame:
     # Create a boolean mask selecting the ROWS where NOT ALL of the columns we
     # care about are invalid (i.e. where ANY of the columns we care about contain a
     # valid value):
-    mask = ~cols_to_check.isin(params.invalid_values).all(axis="columns")
+
+    # for some reason, a simple cols_to_check.isin(params.invalid_values) fails
```
Comment (jdangerx) [inline src/pudl/transform/classes.py:821]: TODO: add specific error + more context from [Slack](https://catalystcooperative.slack.com/archives/C01UQPVETGD/p1694202142945359)

```
TypeError: boolean value of NA is ambiguous
```

Basically this is a fake .isin() because that was running into issues with NA values that we couldn't drop ahead of time.

File: migrations/versions/11a43f756905_idk.py
Comment (jdangerx) [inline migrations/versions/11a43f756905_idk.py:1]: TODO: investigate the purpose of this!

File: migrations/versions/273a78878b74_purchased_storage_mwh.py
Comment (jdangerx) [inline migrations/versions/273a78878b74_purchased_storage_mwh.py:1]: TODO: investigate this one too.

File: src/pudl/analysis/ferc1_eia_train.py
Comment (jdangerx) [inline src/pudl/analysis/ferc1_eia_train.py:1]: TODO: investigate why these changes were made.

**Review by jdangerx — COMMENTED** [review]
Mostly a TODO list for further review / massage.

File: migrations/versions/273a78878b74_purchased_storage_mwh.py
Comment (zaneselvans) [reply migrations/versions/273a78878b74_purchased_storage_mwh.py:1]: I can explain this one

File: migrations/versions/11a43f756905_idk.py
Comment (jdangerx) [reply migrations/versions/11a43f756905_idk.py:1]: This was subsumed by the migration in dev.

File: migrations/versions/273a78878b74_purchased_storage_mwh.py
Comment (jdangerx) [reply migrations/versions/273a78878b74_purchased_storage_mwh.py:1]: Thanks for the context! It looks like this was also fixed via the migration in `dev` so I removed it.

File: src/pudl/analysis/ferc1_eia_train.py
Comment (jdangerx) [reply src/pudl/analysis/ferc1_eia_train.py:1]: This was answered by Austen's commit message - "This also updates the way that values were assigned to a slice of the ferc1_eia_train output spreadsheets. NA values were causing an issue, so I had to change how the values were being converted."

**Comment by jdangerx** [comment]
The integration test failures looked like "there are still some unmapped plants" - so I went ahead and mapped them.

@aesharpe I have a few questions:
* can we assume that `ida grove ii wind farm (8 unites at 2.3 mw each & 73 units at 2.52 mw each)` is the same as `ida grove ii wind farm (8 units @ 2.3 mw each &`, which got cut off in extraction somewhere?
* is `public service company of colorado/conventional hydro` the same as `public service company of colorado/hydro:`?
* when looking at the mt. carmel plants I noticed that `Na` and `not applicable` map to different PUDL IDs, should I map them to the same one?
* should I add plant_name_pudl for all the new plants?

Here's a diff of what the changes are, if they're dumped to CSV:

```diff
diff --git a/pudl_id_mapping_old.csv b/pudl_id_mapping_new.csv
index cdfb14e..628a648 100644
--- a/pudl_id_mapping_old.csv
+++ b/pudl_id_mapping_new.csv
@@ -22800,3 +22800,19 @@ plant_id_pudl,find_plant_id_eia_matches,plant_name_pudl,utility_id_ferc1,utility
 18009,#N/A,total,382,"entergy mississippi, llc",total,,,,,
 18010,#N/A,total,393,"chugach electric association, inc.",total,,,,,
 18011,#N/A,total,444,indiana-kentucky electric corporation,total,,,,,
+18012,#N/A,,153,"pjm interconnection, llc",total,,,,,
+8671,#N/A,,183,pacific gas and electric company,small hydroelectric generating plants:,,,,,
+18013,#N/A,,215,new york state electric & gas corporation,see footnote,,,,new york state electric & gas corporation,
+15000,#N/A,,216,idaho power company,hydro,,,,,
+15001,#N/A,,216,idaho power company,internal combustion,,,,,
+18014,#N/A,,224,"southwest power pool, inc.",total,,,,,
+18015,#N/A,,227,public service company of colorado,community solar gardens,,,,,
+15068,#N/A,,227,public service company of colorado,conventional hydro,,,,,assumed this is the same as just “hydro:”
+17972,#N/A,,249,the empire district electric company,n/a,,,,,
+12926,#N/A,,251,midamerican energy company,ida grove ii wind farm (8 units at 2.3 mw each & 73 units at 2.52 mw each),,,,,
+18016,#N/A,,280,wisconsin electric power company,see footnote,,,,,
+18017,#N/A,,368,upper michigan energy resources company (pudl determined),total,,,,,
+18018,#N/A,,376,"new york transco, llc",total,,,,,
+18019,#N/A,,388,"wilderness line holdings, llc",total,,,,,
+1287,#N/A,,394,alaska electric light and power company,salmon creek hyrdo,,,,,
+18020,#N/A,,397,mt. carmel public utility co,total,,,,,
```

And here are *all* the rows for the plant IDs that actually matched in the new set:

| plant_id_pudl | find_plant_id_eia_matches | plant_name_pudl                          | utility_id_ferc1 | utility_name_ferc1                      | plant_name_ferc1                                                           | plant_id_eia | plant_name_eia                           | utility_id_eia | utility_name_eia               | notes                                     |
|---------------|---------------------------|------------------------------------------|------------------|-----------------------------------------|----------------------------------------------------------------------------|--------------|------------------------------------------|----------------|--------------------------------|-------------------------------------------|
| 1287          | #N/A                      |                                          | 394              | alaska electric light and power company | salmon creek hyrdo                                                         |              |                                          |                |                                |                                           |
| 1287          | #N/A                      | lower salmon creek hydro                 | 394              | alaska electric light and power company | lower salmon creek hydro                                                   |              |                                          |                |                                | 2018 Addition                             |
| 1287          |                           | Salmon Creek 1                           |                  |                                         |                                                                            | 65           | Salmon Creek 1                           | 213            | Alaska Electric Light&Power Co | No FERC ID                                |
| 1287          |                           | Salmon Creek Hydro                       | 394              | Alaska Electric Light and Power Company | Salmon Creek Hydro                                                         |              |                                          |                |                                | "Capacity < 5MW, not mapped"              |
| 1287          | 1287                      | Salmon Creek 1                           | 394              | alaska electric light and power company | upper salmon creek hydro                                                   | 65           | Salmon Creek 1                           | 213            | Alaska Electric Light&Power Co | 2018 Addition                             |
| 12870         |                           | phoenix solar south farms llc solar farm |                  |                                         |                                                                            | 62735        | phoenix solar south farms llc solar farm |                |                                |                                           |
| 12871         |                           | harbec energy                            |                  |                                         |                                                                            | 62736        | harbec energy                            |                |                                |                                           |
| 12872         |                           | west valley west                         |                  |                                         |                                                                            | 62737        | west valley west                         | 59474          | bq energy llc                  |                                           |
| 12873         |                           | west valley east                         |                  |                                         |                                                                            | 62738        | west valley east                         | 59474          | bq energy llc                  |                                           |
| 12874         |                           | kdc solar asgm                           |                  |                                         |                                                                            | 62739        | kdc solar asgm                           |                |                                |                                           |
| 12875         |                           | "msc-empire01, llc"                      |                  |                                         |                                                                            | 62740        | "msc-empire01, llc"                      |                |                                |                                           |
| 12876         |                           | mount joy wire                           |                  |                                         |                                                                            | 62741        | mount joy wire                           |                |                                |                                           |
| 12877         |                           | arcadia solar                            |                  |                                         |                                                                            | 62744        | arcadia solar                            |                |                                |                                           |
| 12878         |                           | wheatridge                               |                  |                                         |                                                                            | 62745        | wheatridge                               |                |                                |                                           |
| 12879         |                           | kdc solar cscp llc                       |                  |                                         |                                                                            | 62746        | kdc solar cscp llc                       |                |                                |                                           |
| 12926         | #N/A                      |                                          | 251              | midamerican energy company              | ida grove ii wind farm (8 units at 2.3 mw each & 73 units at 2.52 mw each) |              |                                          |                |                                |                                           |
| 12926         |                           | ida grove ii                             |                  |                                         |                                                                            | 62795        | ida grove ii                             | 12341          | midamerican energy co          |                                           |
| 12926         | 12926                     | ida grove ii                             | 251              | midamerican energy company              | ida grove ii wind farm (8 units @ 2.3 mw each &                            | 62795        | ida grove ii                             | 12341          | midamerican energy co          |                                           |
| 15000         | #N/A                      |                                          | 216              | idaho power company                     | hydro                                                                      |              |                                          |                |                                |                                           |
| 15000         | #N/A                      | hydro:                                   | 216              | idaho power company                     | hydro:                                                                     |              |                                          |                |                                |                                           |
| 15001         | #N/A                      |                                          | 216              | idaho power company                     | internal combustion                                                        |              |                                          |                |                                |                                           |
| 15001         | #N/A                      | internal combustion:                     | 216              | idaho power company                     | internal combustion:                                                       |              |                                          |                |                                |                                           |
| 15068         | #N/A                      |                                          | 227              | public service company of colorado      | conventional hydro                                                         |              |                                          |                |                                | assumed this is the same as just “hydro:” |
| 15068         | #N/A                      | hydro:                                   | 227              | public service company of colorado      | hydro:                                                                     |              |                                          |                |                                |                                           |
| 17972         | #N/A                      |                                          | 249              | the empire district electric company    | n/a                                                                        |              |                                          |                |                                |                                           |
| 17972         | #N/A                      | total                                    | 249              | the empire district electric company    | total                                                                      |              |                                          |                |                                |                                           |
| 8671          | #N/A                      |                                          | 183              | pacific gas and electric company        | small hydroelectric generating plants:                                     |              |                                          |                |                                |                                           |
| 8671          |                           | Small Hydroelectric Plants:              | 183              | PACIFIC GAS AND ELECTRIC COMPANY        | Small Hydroelectric Plants:                                                |              |                                          |                |                                | "Capacity < 5MW, not mapped"              |

**Comment by jdangerx** [comment]
Re: mt. carmel NA's, here are the relevant rows:

plant_id_pudl | find_plant_id_eia_matches | plant_name_pudl | utility_id_ferc1 | utility_name_ferc1 | plant_name_ferc1
-- | -- | -- | -- | -- | --
8588 |   | Na | 397 | Mt. Carmel Public Utility Co | Na
15030 | #N/A | none | 397 | mt. carmel public utility co | none
15031 | #N/A | not applicable | 222 | mt. carmel public utility co | not applicable
18020 | #N/A | total | 397 | mt. carmel public utility co | total

Note that the FERC utility IDs don't match - they're 222 and 397, which correspond to XBRL entities C000746 and C011301, which actually correspond to these different entities:

```
 dazhong-catalyst@computer ~/p/d/2/ferc1> ag -c 'C000746'                                                                                                                             (pudl-dev) 
Ameren_Illinois_Company_form1_Q4_1681174555.xbrl:4266
dazhong-catalyst@computer ~/p/d/2/ferc1> ag -c 'C011301'                                                                                                                             (pudl-dev) 
Mt._Carmel_Public_Utility_Co._form1_Q4_1680044837.xbrl:459
```

Should we change plant ID 15031 to point at "ameren illinois"?

**Comment by jdangerx** [comment]
OK, this is in the commit message, but I went ahead and committed some changes. @aesharpe let me know if these are reasonable:

Totally new:

* 18012: pjm interconnection, llc / total
* 18013: new york state electric & gas corporation / see footnote
* 18014: southwest power pool, inc. / total
* 18015: public service company of colorado / community solar gardens
* 18016: the empire district electric company / n/a
  each & 73 units at 2.52 mw each)
* 18017: wisconsin electric power company / see footnote
* 18018: upper michigan energy resources company (pudl determined) / total
* 18019: new york transco, llc / total
* 18020: wilderness line holdings, llc / total
* 18021: mt. carmel public utility co / total

Mapped to existing PUDL ID:

* 8671: pacific gas & electric company, small hydroelectric generating plants
* 15000: idaho power company / hydro
* 15001: idaho power company / internal combustion
* 15068: public service company of colorado / conventional hydro
* 12926: midamerican energy company / ida grove ii wind farm (8 units at 2.3 mw
* 1287: alaska electric light and power company / salmon creek hyrdo

Note the misspelling of the plant name in 1287.

Changed:

* 15031: mt. carmel public utility co / not applicable -> ameren
  illinois company / not applicable

  This one had a mismatch between utility_id_ferc 222, which corresponds
  to Ameren, not Mt. Carmel (397).

File: test/conftest.py [hunk 5]
```diff
@@ -263,54 +261,6 @@ def ferc1_xbrl_sql_engine(ferc_to_sqlite_xbrl_only, dataset_settings_config):
...
-                datapackage_path=PudlPaths().output_file(
-                    f"ferc{form.value}_xbrl_datapackage.json"
-                ),
-                metadata_path=PudlPaths().output_file(
-                    f"ferc{form.value}_xbrl_taxonomy_metadata.json"
-                ),
-                archive_file_path=taxonomy_entry_point,
-            )
-
-
```
Comment (zaneselvans) [inline test/conftest.py:313]: Without this fixture, how are the FERC XBRL databases being generated for use in the ETL tests, and how are we doing integration testing to ensure that we're able to extract data from all the forms? Is this just cruft that's been replaced by other fixtures now?

**Review by zaneselvans — APPROVED** [review]
The integration tests are passing so I assume this isn't an issue, but I was confused by the removal of the `ferc_xbrl` fixture. Did it get replaced by something previously, but not ripped out?

File: test/conftest.py (hunk 5 above)
Comment (jdangerx) [reply test/conftest.py:313]: These guys are being generated by the `ferc_to_sqlite_xbrl_only` fixture, now.
